### PR TITLE
Add MVC for pool and team records

### PR DESCRIPTION
**Pull request by wangandrewt**
Add new page for viewing records. Associate records with admin panel.

### CODE REVIEW

File: web/models.py [hunk 1]
```diff
@@ -99,3 +100,122 @@ def has_module_perms(self, app_label):
...
+    event_number = models.PositiveSmallIntegerField()
+    event_name = models.CharField(max_length=100)
+    swimmer_name = models.CharField(max_length=400)
+    swimmer_team = models.CharField(blank=True, max_length=4,
+                                    choices=TEAM_CHOICES)
+    time = models.CharField(max_length=9)
+    date = models.DateField()
+    pool = models.CharField(blank=True, max_length=4, choices=TEAM_CHOICES)
+
+    def __unicode__(self):
```
Comment (y4n9squared) [inline web/models.py]: There are some nasty compatibility issues regarding strings in Python 2.x and Python 3.x. Since we are running Python 2.7 in production, we still need to support byte strings. You can simply add a stub implementation:

```python
def __str__(self):
    return unicode(self).encode('utf-8')
```

Also just a nit: You have some missing spaces between some + operators.

File: web/tests.py [hunk 2]
```diff
@@ -25,6 +25,7 @@
 """
```
Comment (y4n9squared) [inline web/tests.py]: Add yourself to the author list.

File: web/tests.py [hunk 3]
```diff
@@ -103,3 +104,15 @@ def test_coaches_page(self):
...
+
+
+class RecordsPageTestCase(TestCase):
+
+    def test_uses_correct_template(self):
+        url = reverse('records')
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, 'web/records.html')
+
+    def test_records_page(self):
```
Comment (y4n9squared) [inline web/tests.py]: The `assertTemplateUsed` function in Django is a context manager. You can combine both tests like the following:

```python
def test_records_page(self):
    with self.assertTemplateUsed('web/records.html'):
        response = self.client.get('/web/records/')
        self.assertEqual(response.status_code, 200)
```

File: web/views.py [hunk 4]
```diff
@@ -50,6 +53,23 @@ def coaches(request):
...
+    results = SwimRecord.objects.all()
+    team_list = list()
+    pool_list = list()
+    for elem in results:
+        if elem.record_type == SwimRecord.RECORD_TYPE_TEAM:
+            elem.pool_verbose = elem.get_pool_display()
+            team_list.append(elem)
+        else:
+            elem.swimmer_team_verbose = elem.get_swimmer_team_display()
+            pool_list.append(elem)
```
Comment (y4n9squared) [inline web/views.py]: This is starting to muddle the responsibilities of the *view* and the *controller*. It's the controller's job to separate the data (the model). It's the view's job to determine what is "verbose". In general, the model should be considered an **immutable** object representation of the data.

Lines 58-66 can be succinctly achieved via list comprehensions for both clarity and speed.

```python
team_list = [e for e in results if e.record_type == SwimRecord.RECORD_TYPE_TEAM]
pool_list = [e for e in results if e.record_type == SwimRecord.RECORD_TYPE_POOL]
```
Then, in the view, you can choose to display the verbose name:

```html
<td><abbr title="{{ instance.get_pool_verbose }}">{{ instance.pool }}</abbr></td>
```

**Comment by y4n9squared** [comment]
Looking good for the most part. I made a few in-line comments for you to address.

The biggest design concern I have for this is the use of Django to import the CSV into the database. It seems like a roundabout way to load the data when a ton of really nice tools exist to import and export CSV files into tables in SQL databases.

Remember that non-developers are going to be maintaining the site via Django admin and none of them have the faintest clue what a model named 'web.SwimRecord' means. I imagine we will load the data *once* and then use the Django admin interface for the SwimRecord model to make any necessary changes from that point forward.

In summary, I don't think we should be using `django-csvimport`. It doesn't seem like the right solution to our problem.
